lib/filterx/filterx-pipe.c

```diff
@@ -66,9 +66,22 @@ log_filterx_pipe_queue(LogPipe *s, LogMessage *msg, const LogPathOptions *path_o
   FilterXEvalContext eval_context;
   LogPathOptions local_path_options;
   FilterXEvalResult eval_res;
+  gboolean local_scope = FALSE;
 
   path_options = log_path_options_chain(&local_path_options, path_options);
-  filterx_eval_init_context(&eval_context, path_options->filterx_context, msg);
+
+  FilterXScope *scope = NULL;
+  if (path_options->filterx_context)
+    scope = filterx_scope_reuse(path_options->filterx_context->scope);
+
+  if (!scope)
+    {
+      gsize alloc_size = filterx_scope_get_alloc_size();
```

Comment:
Now that we are on the stack, we should definitely put an upper limit on the maximum allocated variables count.